desktop/ui/src/main/java/org/datacleaner/panels/ExecuteJobWithoutAnalyzersDialog.java

```diff
@@ -143,12 +144,19 @@ public void actionPerformed(ActionEvent e) {
 
                 final AnalyzerComponentBuilder<? extends Analyzer<?>> analyzer = copyAnalysisJobBuilder
                         .addAnalyzer(analyzerClass);
+
                 analyzer.addInputColumns(copyAnalysisJobBuilder.getAvailableInputColumns(Object.class));
 
                 final String formattedDate = new SimpleDateFormat("yyyy-MM-dd").format(new Date());
 
-                analyzer.setConfiguredProperty("File",
-                        createFile("datacleaner-" + formattedDate + "-output", filenameExtension));
+                final FileResource resource = createResource("datacleaner-" + formattedDate + "-output",
+                        filenameExtension);
+                if (filenameExtension.equals(".xls") || filenameExtension.equals(".xlsx")) {
```

Comment:
This is quite unclear why the "if" is here. I imagine it's because the CreateExcelAnalyzer is still using a java.io.File? If so then construct the if around that fact so that it's clear:

if (analyzer.getComponentClass() == CreateExcelAnalyzer.class) { ...